binrw_derive/src/binrw/codegen/read_options/enum.rs

```diff
@@ -129,18 +129,30 @@ fn generate_unit_enum_magic(reader_var: &TokenStream, variants: &[UnitEnumField]
                 #body
             })() {
                 v @ Ok(_) => return v,
-                Err(#TEMP) => { #RESTORE_POSITION_VARIANT(#reader_var, #POS, #TEMP)?; }
+                Err(#TEMP) => {
+                    if !#RESTORE_POSITION_VARIANT(#reader_var, #POS, #TEMP)?.is_eof() {
```

Comment:
This can be expressed a little more concisely as `#ALL_EOF &= #TEMP.is_eof();`. Using the error passed through `#RESTORE_POSITION_VARIANT` is theoretically slightly more efficient, but only in the error case, which is the slow path anyway, so it’s probably better to just use `#TEMP` directly for clarity and have two statements instead of one big one.

Reply:
Thanks for the details! Your version is much cleaner 👍